src/eei.cpp

```diff
@@ -603,6 +679,7 @@ namespace HeraVM {
 
       evm_address address = loadUint160(addressOffset);
 
+      takeGas(GasSchedule::selfdestruct + (!context->fn_table->account_exists(context, &address)) ? GasSchedule::callnewacct : 0);
```

Comment:
I'd split this up into two `takeGas` statements for readability.

Reply:
done